felix/ipsets/ipsets_test.go

```diff
@@ -464,6 +466,42 @@ var _ = Describe("IP sets dataplane", func() {
 		})
 	})
 
+	It("Calico IP sets with unsupported revision and in desired sets should be re-created", func() {
```

Comment:
Please add the counterpoint test: if everything is correct then it should not do a swap.  Doing a swap will always result in the right answer but it's a lot slower than re-using what's there; it invalidates a major optimisation.

Reply:
Added one in the same test.